include/zephyr/bluetooth/audio/hap.h

```diff
@@ -0,0 +1,394 @@
+	 * @param conn Controller instance pointer.
+	 * @param err 0 in case of success or negative errno or ATT Error code in case of error.
+	 */
+	void (*connected)(struct bt_hap_harc *harc, int err);
+
+	/**
+	 * @brief Controller has been disconnected.
+	 *
+	 * This callback notifies the application that the controller has been disconnected.
+	 *
+	 * @param conn Controller instance pointer.
+	 */
+	void (*disconnected)(struct bt_hap_harc *harc);
+
+	/**
+	 * @brief Controller info has been updated.
+	 *
+	 * This callback notifies the application that the controller info has been changed,
+	 * e.g. the Hearing Aid Features has been updated.
+	 *
+	 * @param conn Controller instance pointer.
+	 * @param info Info object.
+	 */
+	void (*info)(struct bt_hap_harc *harc, const struct bt_hap_harc_info *info);
+
+	sys_snode_t _node;
+};
+
+/**
+ * @brief Register Hearing Aid Remote Controller callbacks.
+ *
+ * @param cb The callback structure.
+ *
+ * @return 0 if success, errno on failure.
+ */
+int bt_hap_harc_cb_register(struct bt_hap_harc_cb *cb);
+
+/**
+ * @brief Unregister Hearing Aid Remote Controller callbacks.
+ *
+ * @param cb The callback structure.
+ *
+ * @return 0 if success, errno on failure.
+ */
+int bt_hap_harc_cb_unregister(struct bt_hap_harc_cb *cb);
+
+/**
+ * @brief Bind profile.
+ *
+ * Bind profile to remote device implementing HAP Hearing Aid role.
+ *
+ * @param conn Bluetooth connection object.
+ * @param[out] harc Valid controller object on success.
+ *
+ * @return Zero on success or (negative) error code on failure.
+ */
+int bt_hap_harc_bind(struct bt_conn *conn, struct bt_hap_harc **harc);
+
+/**
+ * @brief Unbind profile.
+ *
+ * Unbind profile or cancel pending profile binding.
+ *
+ * @param harc Controller instance pointer.
+ *
+ * @return Zero on success or (negative) error code on failure.
+ */
+int bt_hap_harc_unbind(struct bt_hap_harc *harc);
+
+/** @brief Hearing Aid Remote Controller Preset List callback structure. */
+struct bt_hap_harc_preset_cb {
+	/**
+	 * @brief Callback function for Hearing Access Service active preset changes.
+	 *
+	 * Optional callback called when the active preset is changed by the remote server when the
+	 * preset switch procedure is complete. The callback must be set to receive active preset
+	 * changes and enable support for switching presets.
+	 *
+	 * @param harc Controller instance pointer.
+	 * @param index Active preset index.
+	 */
+	void (*active)(struct bt_hap_harc *harc, uint8_t index);
+
+	/**
+	 * @brief Callback function for preset store.
+	 *
+	 * The callback is called when the Preset Record Update ot Preset Read Response is notified
+	 * by the remote server.
+	 * The record object as well as its objects are temporary and must be copied to in order
+	 * to cache its information.
+	 *
+	 * @param harc Controller instance pointer.
+	 * @param record Preset record to store.
+	 */
+	void (*store)(struct bt_hap_harc *harc, const struct bt_has_preset_record *record);
```

Comment:
Unsure I understand what this does. 
Is this the opposite to `remove`?

Reply:
Yes, this application callback is called to request the app to store the preset record (or update the existing entry if it's already there)

Reply:
Alright, that makes sense. Not completely sold on the naming of this function though. 

Since it's handling a notification/read of a preset, maybe `(*preset)` could work? Or is that too generic? 

I would also be OK with the current name, so not a blocker.